db/R/query.file.path.R

```diff
@@ -0,0 +1,19 @@
+##' @name db.query.file.path
+##' @title Get file path given id and machine
+##' @param input_id
+##' @param host_name
+##' @param con : database connection
+##' @export 
+##' 
+##' @author Betsy Cowdery 
+query.file.path <- function(input.id, host_name, con){
+  machine.host <- ifelse(host$name == "localhost",fqdn(),host_name)
+  machine = db.query(paste0("SELECT * from machines where hostname = '",machine.host,"'"),con)
+  dbfile = db.query(paste("SELECT file_name,file_path from dbfiles where container_id =",input.id," and container_type = 'Input' and machine_id =",machine$id),con)
+  path <- file.path(dbfile$file_path,dbfile$file_name)
+  if(file.exists(path)){
```

Comment:
checking if file exists would depend on the user having permissions to
access the file. I may wish to know about a file even if I don't have
permission to use it ... the most pertinent example is that I may want to
ask for permission if a file is available somewhere rather than moving my
own copy

On Tue, Nov 10, 2015 at 8:45 PM, Betsy Cowdery <notifications@github.com>
wrote:

> In db/R/query.file.path.R
> <https://github.com/PecanProject/pecan/pull/674#discussion_r44496489>:
>
> > @@ -0,0 +1,19 @@
> > +##' @name db.query.file.path
> > +##' @title Get file path given id and machine
> > +##' @param input_id
> > +##' @param host_name
> > +##' @param con : database connection
> > +##' @export
> > +##'
> > +##' @author Betsy Cowdery
> > +query.file.path <- function(input.id, host_name, con){
> > +  machine.host <- ifelse(host$name == "localhost",fqdn(),host_name)
> > +  machine = db.query(paste0("SELECT * from machines where hostname = '",machine.host,"'"),con)
> > +  dbfile = db.query(paste("SELECT file_name,file_path from dbfiles where container_id =",input.id," and container_type = 'Input' and machine_id =",machine$id),con)
> > +  path <- file.path(dbfile$file_path,dbfile$file_name)
> > +  if(file.exists(path)){
>
> Should this be checked with remote.execute?
> Something like:
>
> cmd <- paste("R file.exists(",path,")")
> remote.execute.R(cmd,machine.host,verbose=TRUE)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PecanProject/pecan/pull/674/files#r44496489>.
>